jetbrains-core/src/software/aws/toolkits/jetbrains/core/AwsResourceCache.kt

```diff
@@ -291,12 +294,25 @@ class DefaultAwsResourceCache(
 
     private fun <T> getCachedResource(context: Context<T>) {
         ApplicationManager.getApplication().executeOnPooledThread {
+            var currentValue: Entry<T>? = null
             try {
                 @Suppress("UNCHECKED_CAST")
                 val result = cache.compute(context.cacheKey) { _, value ->
```

Comment:
Why is this compute and not computeIfAbsent?

Reply:
we could rewrite as `computeIfAbsent`. the logic right now re-inserts the same value if the cached value is still good, otherwise it kicks off a refresh.